### PR TITLE
organizing the solver options

### PR DESCRIPTION
**Pull request by saeid3**
This addresses the comments of this PR https://github.com/jvail/glpk.js/pull/17

### CODE REVIEW

File: glpk.d.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ interface LP {
     }[],
     binaries?: string[],
     generals?: string[],
-    settings?: { mipGap?: number, tmLim?: number, msgLev?: number }
+    options?: { mipGap?: number, tmLim?: number, msgLev?: number, preSolve?: number }
```
Comment (rregue) [inline glpk.d.ts]: for this comment, I think we need to declare a new interface, like:

```
interface Options {

}
```
and then reference this interface here and below on the solve.

File: src/pre.js [hunk 2]
```diff
@@ -94,13 +95,16 @@ var glpkPromise = new Promise(function (resolve) {
...
+				'solve': function (lp, opt) {
 					
-					const settings = _.defaults(options, {
-							msgLev: 1,
-							tmLim: 120000,
-							mipGap: 0
-					})
+					const opt_ = opt || lp.options || {};
+					
+			    const options = {
```
Comment (rregue) [inline src/pre.js]: the indentation here does not seem right

File: src/pre.js [hunk 3]
```diff
@@ -94,13 +95,16 @@ var glpkPromise = new Promise(function (resolve) {
...
+			        preSolve: typeof opt_.preSolve !== 'undefined' ? +(!!opt_.preSolve) : 1,
```
Comment (rregue) [inline src/pre.js]: I think presolve is one word, so maybe no need to `preSolve` and instead `presolve`

**Review by rregue — CHANGES_REQUESTED** [review]
Left a few comments, I think one is blocking which is the one around creating the interface for the options object.

File: src/pre.js (hunk 2 above)
Comment (saeid3) [reply src/pre.js]: hmm it was a combination of tabs & whitespace that messed up

File: src/pre.js (hunk 3 above)
Comment (saeid3) [reply src/pre.js]: it WAS `presolve` before, and I reflected that resolve is of course two words and I should change!

**Review by rregue — APPROVED** [review]
👍

**Review by rregue — APPROVED** [review]
👍

**Pull request merged**